src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            increaseSize();
+        }
+        array[size] = value;
+        ++size;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        indexCheck(index);
```

Comment:
let's add checking capacity as well
because what if we don't have place to put the element?